hoot-core/src/main/cpp/hoot/core/ops/BuildingOnlyOp.cpp

```diff
@@ -76,7 +76,7 @@ void BuildingOnlyOp::apply(boost::shared_ptr<OsmMap>& map)
   // Visit the map once, execute all visitors on each element
   MultiVisitor multiVtor;
   multiVtor.addVisitor(&removeTagVtor);
-  multiVtor.addVisitor(&replaceTagVtor);
+  multiVtor.addVisitor(reinterpret_cast<ElementOsmMapVisitor *>(&replaceTagVtor));
   multiVtor.addVisitor(reinterpret_cast<ElementOsmMapVisitor *>(&removeEVisitor));
   map->visitRw(multiVtor);
```

Comment:
MultiVisitor is feeding FilteredVisitor at some point, which gets called with the OsmMap visit method.  So, there's no choice but to use ElementOsmMapVisitor here, despite the fact some of the child visitors may not need to use a map. ..thus the adding of another cast.  I'm going to try to make changes later so that ElementVisitor can be used here instead when OsmMap consumption isn't required.